simphony_ui/ui.py

```diff
@@ -197,12 +211,17 @@ def _run_calc_threaded(self):
         """ Function which will run the calculation. This function
         is only run by the secondary thread
         """
-        return run_calc(
-            self.global_settings,
-            self.openfoam_settings,
-            self.liggghts_settings,
-            self.update_progress_bar
-        )
+        try:
+            return run_calc(
+                self.global_settings,
+                self.openfoam_settings,
+                self.liggghts_settings,
+                self.update_progress_bar
+            )
+        except Exception as e:
+            self.calculation_error_event = str(e)
```

Comment:
This is OK as-is, but a better approach would be to pass both the string of the exception _and_ the string of the traceback (you can get this via the `traceback` module) as a tuple, then you can supply the traceback as the `detail` in the error dialog, which can help with end-user debugging.

This isn't critical for a small app like this, particularly since you are also logging the exception.